packages/mst-middlewares/src/undo-manager.ts

```diff
@@ -135,7 +135,11 @@ const UndoManager = types
                 self.undoIdx = self.history.length
             },
             afterCreate() {
-                targetStore = getEnv(self).targetStore ? getEnv(self).targetStore : getRoot(self)
+                try {
+                    targetStore = getEnv(self).targetStore
+                } catch {
+                    targetStore = getRoot(self)
```

Comment:
This is an expensive and ugly way to check for the existence of an environment. I think we should introde `hasEnv(self)` to accompany this change

Reply:
I introduced `hasEnv`, do you want to have a look at that function ?